app/views/assignment/edit.html.erb

```diff
@@ -14,4 +9,9 @@
     <%= form.hidden_field :week_id, :value => @week.id %>
     <%= form.submit "Save Changes" %>
   <% end %>
-<% end %>
\ No newline at end of file
+<% end %>
+
+
+<% if @assignments.count < 2 %>
+  <div><%= link_to "Add assignment", new_assignment_path(:week => @week) %></div>
```

Comment:
I just realized that in some places you're using the hash rocket syntax (`:week => @week`) and in other places you're using the 'new' syntax (`week: @week`). It's not a big deal here but we strictly use the 'new' syntax in other projects and I imagine that most newer Rails projects are the same.